lib/errors.dart

```diff
@@ -89,6 +89,9 @@ enum PurchasesErrorCode {
   /// This is a required property and likely an AppStore quirk that it is missing.
   productDiscountMissingIdentifierError,
 
+  /// Unknown error
+  unknownNonNativeError,
```

Comment:
I added a new error so,

I thought adding a new error was better than just processing the error code in PHC with the missing error, since if any dev was using the enum indexes for this (which they shouldn't) it would still match the error codes. This new error should never happen though. Let me know what you think!